docs/docs/guides/05-type-system/02-concepts/02-uid.mdx

```diff
@@ -0,0 +1,188 @@
+
+:::note
+ContentType and Component are referring to both the related schema and entity.
+:::
+
+|                          |    ContentType     |     Component      |     Middleware     |       Policy       |     Controller     |      Service       |
+|--------------------------|:------------------:|:------------------:|:------------------:|:------------------:|:------------------:|:------------------:|
+| `api::<scope>.<name>`    | :white_check_mark: |        :x:         | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: |
+| `plugin::<scope>.<name>` | :white_check_mark: |        :x:         | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: |
+| `<category>.<name>`      |        :x:         | :white_check_mark: |        :x:         |        :x:         |        :x:         |        :x:         |
+| `strapi::<name>`         | :white_check_mark: |        :x:         | :white_check_mark: | :white_check_mark: |        :x:         |        :x:         |
+| `admin::<name>`          | :white_check_mark: |        :x:         | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: |
+| `global::<name>`         |        :x:         |        :x:         | :white_check_mark: | :white_check_mark: |        :x:         |        :x:         |
+
+### Usage
+
+When referencing resource by their UID you'll need to use the `UID` namespace exported from `@strapi/types`.
+
+```typescript
+import type { UID } from '@strapi/types';
+```
+
+This namespace contains shortcuts to dynamic UID types built from the public registries so that they always adapt to the current context.
+
+:::danger
+The `UID` namespace is designed to be the main interface used by developers.
+
+**Do not** use the `Internal.UID` namespace **except if you know what you're doing** (low level extends clause, isolated internal code, etc...).
+:::
+
+#### Basic Example
+
+A common usage is to declare a function that takes a UID as parameter.
+
+For our example, let's imagine we want to fetch an entity based on the provided resource UID.
+
+```typescript
+import type { UID, Data } from '@strapi/types';
+
+declare function fetch(uid: UID.ContentType): Data.ContentType;
+```
+
+:::tip
+To find an exhaustive list of available UID types, take a look at the [related API reference](http://foo)
+:::
+
+#### Parameter Type Inference
+
+Now let's say we want to adapt the return type of our function, so that it matches the given UID.
+
+```typescript
+fetch('api::article.article');
+// ^ this should return a Data.Entity<'api::article.article'>
+
+fetch('admin::user');
+// ^ this should return a Data.Entity<'admin::user'>
+```
+To do that, we'll need the function to be able to provide us with the current `uid` type based on usage.
+
+```typescript
+import type { UID, Data } from '@strapi/types';
+
+declare function fetch<T extends UID.ContentType>(uid: T): Data.ContentType<T>;
```

Comment:
Based on the below warning about not using T as a type variable, would it be wiser to give an example that is good/random and functional? Like... rabbits or trainers or something? 😁

Reply:
Yeah, I wasn't sure there. At first I made the example with a good variable name, but it was a bit bloated and was taking the focus away from the types I wanted to highlight. Quite unsure about what's the best option here :confused: